src/robot/api/deco.py

```diff
@@ -96,9 +124,38 @@ def decorator(func):
     return decorator
 
 
+C = TypeVar('C')
+
+
+# Decorator used without arguments, i.e. @library()
+@overload
+def library(cls: Type[C], /) -> Type[C]:
+    ...
+
+
+# Decorator used with arguments, e.g. @library(version="1.0.0")
+@overload
+def library(
+    scope: 'str | None' = None,
+    version: 'str | None' = None,
+    converters: 'Converters | None' = None,
+    doc_format: 'str | None' = None,
+    listener: 'Type[Any] | None' = None,
+    auto_keywords: bool = False,
+) -> Callable[[Type[C]], Type[C]]:
```

Comment:
Needing to use `Type[C]` everywhere is a bit annoying. I believe that if the `TypeVar` is bound to `type` that isn't needed. At least it seems to work fine when I tested it locally.